lib/source/s3.js

```diff
@@ -0,0 +1,284 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const Aws = require('aws-sdk');
+
+const DEFAULT_INTERVAL = 60000;
+
+class S3Parser {
+  constructor() {
+    this.properties = {};
+  }
+
+  update(data) {
+    let properties = {};
+
+    try {
+      properties = JSON.parse(data.toString());
+    } catch (e) {
+      // We should have an error condition but tbd
```

Comment:
I didn't see a test case for this `catch` branch, but I might have just missed it.